File: services/interaction-service/src/main/java/com/socialmedia/interactionservice/dto/CommentDto.java
package com.socialmedia.interactionservice.dto;

import lombok.*;

@Builder
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
public class CommentDto {
    private Long id;
    private Long postId;
    private String userId;
    private String firstname;
    private String lastname;
    private String content;
    private Long parentCommentId;
    private Boolean isEdited;
    private Long likeCount;
    private Long replyCount;
    private String createdAt;
    private String updatedAt;
}

File: client/src/app/shared/models/comments/comment.dto.ts
export type CommentDto = {
  id: number;
  postId: number;
  userId: string;
  firstname: string;
  lastname: string;
  username: string;
  profilePictureUrl?: string;
  content: string;
  parentCommentId: number | null;
  isEdited: boolean;
  likeCount: number;
  replyCount: number;
  createdAt: string;
  updatedAt: string;
};
